pyfr/denserkschemes/__init__.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+
+from pkg_resources import resource_listdir, resource_string
+import re
+
+import numpy as np
+
+
+class BaseTabulatedDensePseudoStepper(object):
+    def __init__(self, rule):
+        a = []
+
+        rule = re.sub(r'(?<=\))\s*,?\s*(?!$)', r'\n', rule)
+        rule = re.sub(r'\(|\)|,', '', rule).strip()
+        rule = rule[1:-1] if rule.startswith('[') else rule
+
+        for l in rule.splitlines()[:-1]:
+            if not l:
+                continue
+
+            # Store the Butcher A matrix
+            a.append([float(f) for f in l.split()])
+
+        # The last line is the b vector
+        b = [float(f) for f in rule.splitlines()[-1].split()]
+
+        # Cast
+        self.a = np.array(a, dtype=np.float)
+        self.b = np.array(b, dtype=np.float)
+
+
+class BaseDensePseudoStepper(BaseTabulatedDensePseudoStepper):
```

Comment:
Should be general enough to handle a random dense scheme (no p-optimal variants).